### PR TITLE
Adding pyi files for better docs

### LINKED ISSUE #1187: Use a custom auto-generated typing class to add good docs to the python client
The idea is the run a tox command, that:

- Reads the .proto file comments, types etc and generate a `typings.py` file with descriptions of each type. The comments in the .proto files can include links to relavent docs.
- These types would not be truly connected to the grpc generated files, but instead be user readable, created to the standards we use in our codebase. 
- The types could then be implemented into `client.py` to give extra infomation.

### PR DESCRIPTION
**Pull request by Sharpz7**
Closes #1187 

`.pyi` files allow us to have seperate files to the grpc generated ones that are signifcantly more readable. For example:

```py
class JobSubmittedEvent(google.protobuf.message.Message):
    DESCRIPTOR: google.protobuf.descriptor.Descriptor
    JOB_ID_FIELD_NUMBER: builtins.int
    JOB_SET_ID_FIELD_NUMBER: builtins.int
    QUEUE_FIELD_NUMBER: builtins.int
    CREATED_FIELD_NUMBER: builtins.int
    JOB_FIELD_NUMBER: builtins.int
    job_id: typing.Text
    job_set_id: typing.Text
    queue: typing.Text
    @property
    def created(self) -> google.protobuf.timestamp_pb2.Timestamp: ...
    @property
    def job(self) -> armada.queue_pb2.Job: ...
    def __init__(self,
        *,
        job_id: typing.Text = ...,
        job_set_id: typing.Text = ...,
        queue: typing.Text = ...,
        created: typing.Optional[google.protobuf.timestamp_pb2.Timestamp] = ...,
        job: typing.Optional[armada.queue_pb2.Job] = ...,
        ) -> None: ...
    def HasField(self, field_name: typing_extensions.Literal["created",b"created","job",b"job"]) -> builtins.bool: ...
    def ClearField(self, field_name: typing_extensions.Literal["created",b"created","job",b"job","job_id",b"job_id","job_set_id",b"job_set_id","queue",b"queue"]) -> None: ...
global___JobSubmittedEvent = JobSubmittedEvent
```

There is nothing close to this level of detail in the generated stub files. This is extremely useful for accessing the exact type of different objects, and by using mypy you also get error detection. 

Here is an example in vscode using `mypy`, hovering over a `JobSubmitRequestItem` object.

## With the .pyi files

As you can see there is significantly more infomation.

### CODE REVIEW

File: scripts/build-python-client.sh [hunk 1]
```diff
@@ -6,10 +6,11 @@ mkdir -p proto/armada
 cp pkg/api/event.proto pkg/api/queue.proto pkg/api/submit.proto pkg/api/usage.proto proto/armada
 sed -i 's/\([^\/]\)pkg\/api/\1armada/g' proto/armada/*.proto
 
+pip3 install mypy-protobuf
```
[the remark this answers is not in the document]
Comment (Sharpz7) [reply scripts/build-python-client.sh:9]: On it :))

**Comment by Sharpz7** [comment]
@kannon92 any suggestions as to why why machine messed with the lockfile so much in this commit?

**Comment by kannon92** [comment]
> @kannon92 any suggestions as to why why machine messed with the lockfile so much in this commit?

None. @jayofdoom and I are going to switch to setup tools so this will probably go away.  

I think w/e commit you have is fine.

**Review by kannon92 — DISMISSED** [review]

**Review by kannon92 — APPROVED** [review]

**Review by jayofdoom — APPROVED** [review]

**Review by kannon92 — APPROVED** [review]

**Review by dejanzele — APPROVED** [review]

**Comment by Sharpz7** [comment]
Okay, I'll do that within the next 2 hours

**Pull request merged**